### ISSUE TITLE
Shift clicking a pearl into a furnace breaks tracking

### ISSUE THREAD

**Issue by Lazersmoke** [DevotedMC/ExilePearl#33]
This happens even when the furnace is already full. The pearl stays in your inventory, but `/ep check` will show it as being in the furnace, or in some cases, a block at the coordinates of the furnace.

**Comment by ProgrammerDan** [comment]
makes sense; different tile entities trigger different types of event triggers. I would recommend adding some debug code to output those Enums, and match against the logic in tracking, to see what needs improving.

**Comment by Lazersmoke** [comment]
Is there a way to determine ahead of time if something will actually get placed into an inventory slot during a `PLACE_` InventoryClickEvent? For example, attempting to place a pearl into a furnace's fuel slot temporarily breaks tracking (fixes itself when you place it in your inventory or drop it, but will get freed if you `/ep locate` while it's in your cursor) because it assumes that it was successfully placed, while it actually doesn't go in the slot because it's a fuel slot :/

**Comment by ProgrammerDan** [comment]
_so far_ in my testing of this nature... no. You basically need increasingly complex checks to decide if something will happen, or a post-check to validate that it _did_ happen ... none of which is really supported and much of which is vulnerable to "rapid clicks" or other stupidly degenerate outcomes due to how minecraft handles things and as a consequence how spigot wraps and exposes it.

Simplest route is just prevent placement in all but a select few containers, then fully enumerating the containers you want to support.

**Comment by Lazersmoke** [comment]
I think it is sufficient to limit it to only inventories that can accept every item in every slot, so hoppers, chests, double chests, dispensers, droppers.

**Comment by ProgrammerDan** [comment]
That should simply the logic a good deal.
